gym_minigrid/oracle_agent.py

```diff
@@ -147,19 +147,17 @@ def generate_demos(self, num_demos=1):
             obss, rewards, actions = [], [], []
             obs, target = self.reset()
             mission = obs["mission"]
-            obss.append(obs)
-            rewards.append(0)
             if self.visualize:
                 # Blocking event loop
                 self.window.show(block=False)
 
             for action in self.get_sequence(target):
                 if action is None:
                     break
-                actions.append(action)
-                obs, reward, done, info = self.step(action)
                 obss.append(obs)
-                rewards.append(0)
+                obs, reward, done, info = self.step(action)
+                actions.append(action)
+                rewards.append(reward)
```

Comment:
In the updated code don't we lose the final observation? I think it should be:

```python
obss.append(obs) # append initial state
for action in self.get_sequence(target):
    if action is None:
        break
    obs, reward, done, info = self.step(action)
    obss.append(obs)
    actions.append(action) # actions and rewards are offset from state by 1, i.e. action 0 moved state from 0->1
    rewards.append(reward)
```

Also before the rewards and observations were the same size and now they will be offset by 1. What is the expected behaviour here? I would have assumed that we would want `len(actions)==len(rewards)==len(obss)+1` but that's not what was in the old code it seems.